client/mumble-plugin/lib/audio.cpp

```diff
@@ -52,9 +67,39 @@ void fgcom_audio_addNoise(float noiseVolume, float *outputPCM, uint32_t sampleCo
 void fgcom_audio_applyVolume(float volume, float *outputPCM, uint32_t sampleCount, uint16_t channelCount) {
     // just loop over the array, applying the volume
     if (volume == 1.0) return; // no adjustment requested
-    // TODO: Make sure we are not going off limits
+    
+    // Clamp volume to safe range to prevent audio distortion
+    if (volume < 0.0) volume = 0.0;
+    if (volume > 2.0) volume = 2.0;  // Allow some headroom but prevent excessive amplification
```

Comment:
Just an Idea. It might be good to have a log message, in case the clamp was exceeded.
I think, that this should be no normal behaviour. But it might. If this arises, needs separate investigation.

```
    // Clamp volume to safe range to prevent audio distortion
    if (volume < 0.0) {
        volume = 0.0;
        pluginDbg("ATTENTION: volume < 0.0 (clamped) in fgcom_audio_applyVolume()");
    }
    if (volume > 2.0) {
        volume = 2.0;  // Allow some headroom but prevent excessive amplification
        pluginDbg("ATTENTION: volume > 2.0 (clamped) in fgcom_audio_applyVolume()");
    }
```